uaclient/tests/test_lock.py

```diff
@@ -0,0 +1,219 @@
+import mock
+import pytest
+
+from uaclient.exceptions import LockHeldError
+from uaclient.lock import SingleAttemptLock, SpinLock
+
+M_PATH = "uaclient.lock."
+M_PATH_UACONFIG = "uaclient.config.UAConfig."
+
+
+@mock.patch("os.getpid", return_value=123)
+@mock.patch(M_PATH_UACONFIG + "delete_cache_key")
+@mock.patch(M_PATH_UACONFIG + "add_notice")
+@mock.patch(M_PATH_UACONFIG + "write_cache")
+class TestSingleAttemptLock:
+    def test_creates_and_releases_lock(
+        self,
+        m_write_cache,
+        m_add_notice,
+        m_delete_cache_key,
+        _m_getpid,
+        FakeConfig,
+    ):
+        cfg = FakeConfig()
+        arg, kwarg = mock.sentinel.arg, mock.sentinel.kwarg
+
+        def test_function(args, cfg):
+            assert arg == mock.sentinel.arg
+            assert kwarg == mock.sentinel.kwarg
+            return mock.sentinel.success
+
+        with SingleAttemptLock(cfg, "some operation"):
+            ret = test_function(arg, cfg=cfg)
+
+        assert mock.sentinel.success == ret
+        assert [
+            mock.call("lock", "123:some operation")
+        ] == m_write_cache.call_args_list
+        lock_msg = "Operation in progress: some operation"
+        assert [mock.call("", lock_msg)] == m_add_notice.call_args_list
+        assert [mock.call("lock")] == m_delete_cache_key.call_args_list
+
+    def test_creates__and_releases_lock_when_error_occurs(
```

Comment:
I think we can remove the extra `_` between `creates` and `and`

Reply:
:facepalm: